# LABORATORIO DIRIGIDO N.° 02
## Analítica Empresarial Integrada — TECSUP 2026-II

| | |
|---|---|
| **Docente** | Pilar Rocío Sayán Mejía |
| **Laboratorio** | Perfil analítico comparado de empresas del sector azucarero peruano |
| **N.° de laboratorio** | 2 |
| **Tema** | Estadios de evolución analítica y modelo DELTA de madurez organizacional |
| **Plan curricular** | Big Data y Ciencia de Datos · Código C28 · 2026-II |
| **Duración** | 190 minutos (7:00 p.m. – 10:10 p.m., con receso de 15 minutos) |
| **Fuente de datos** | SMV y BCRP — datos reales y oficiales |
| **Modalidad** |

---

**Equipo N.°:** 2 &nbsp;&nbsp;&nbsp; **Sección:** C28 &nbsp;&nbsp;&nbsp; **Fecha:** 28/08/2026

**Apellidos y nombres del estudiante:** _(Arteaga Guerra Pedro Sebastian Alfieri)_

**Integrantes del equipo:** _(apellidos y nombres de todos los integrantes)_

---


**Elemento de la capacidad terminal:** comprende los estadios de evolución analítica y las dimensiones del modelo DELTA y de su ampliación DELTA Plus; aplica estos marcos a la comparación de indicadores financieros reales y diferencia evidencia, inferencia y opinión.

> **Alcance.** Se analizan estados financieros individuales y anuales del ejercicio 2024. Los importes de la SMV se expresan en miles y pueden estar reportados en soles o dólares según la empresa. Las razones financieras son comparables cuando el numerador y el denominador pertenecen a la misma empresa, moneda y periodo.


## Organización de la sesión — 7:00 p. m. a 10:10 p. m.

**Duración total:** 190 minutos · **Receso:** 15 minutos · **Trabajo efectivo:** 175 minutos.

| Horario | Tiempo | Desarrollo |
|---|---:|---|
| 7:00–7:10 | 10 min | Apertura del caso y conexión con los estadios de evolución analítica. |
| 7:10–7:35 | 25 min | Actividad 1. Revisión de conceptos. Modelos DELTA y DELTA Plus. |
| 7:35–8:00 | 25 min | Actividad 2. Entorno y descarga de información financiera de la SMV. |
| 8:00–8:30 | 30 min | Actividad 2. Normalización, validación e indicadores financieros. Reto 1. |
| 8:30–8:45 | 15 min | **RECESO** |
| 8:45–9:15 | 30 min | Actividad 2. Comparación con Polars y DuckDB. Visualización. Reto 2. |
| 9:15–9:35 | 20 min | Actividad 2. Contexto sectorial del BCRP. Reto 3. |
| 9:35–10:00 | 25 min | **Reto de aplicación y retroalimentación.** Ejercicios 1 a 5. |
| 10:00–10:10 | 10 min | Actividad 3. Informe, conclusiones y preguntas de DELTA Plus. |

> **Regla de trabajo:** no avance de bloque sin registrar la interpretación solicitada. El objetivo no es ejecutar celdas, sino convertir datos en evidencia para una decisión.


**Actividad 1 — Revisión de conceptos (25 minutos)**

**Propósito.** Consolidar el vocabulario técnico de la semana antes del desarrollo práctico. La precisión conceptual constituye un requisito del análisis: un indicador correctamente calculado pierde valor si se interpreta con categorías imprecisas.

**Instrucciones.** Complete la tabla con definiciones elaboradas con sus propias palabras. No se admite la reproducción literal de fuentes externas ni de sistemas generativos. La columna de la derecha contiene una pregunta de apoyo: si su definición permite responderla, la definición es suficiente; si no lo permite, corríjala antes de continuar.

**Evidencia esperada.** Tabla completa con las definiciones registradas.

| Concepto | Definición elaborada por el estudiante | Pregunta de apoyo |
|---|---|---|
| Estadios de la evolución analítica | Fases de madurez en el uso de datos. Una empresa en estadio 2 usa analítica de forma aislada por departamentos, mientras que en el estadio 3 la visión y los recursos se integran a nivel de toda la organización. | ¿Qué distingue a una organización del estadio 2 de una del estadio 3? |
| Modelo DELTA | Marco de madurez basado en Datos, Empresa, Liderazgo, Objetivos y Analistas. Avanzar en conjunto es vital porque tener excelentes analistas sin el liderazgo que los respalde o los datos correctos hace que las iniciativas fracasen. | ¿Por qué el modelo sostiene que las dimensiones deben avanzar de manera conjunta? |
| Modelo DELTA Plus | Evolución del marco original que añade Tecnología y Técnicas Analíticas. Corrige la limitación de ignorar el rol habilitador de infraestructuras modernas (como la nube) y algoritmos avanzados en la estrategia. | ¿Qué limitación del modelo original corrige la ampliación? |
| Dimensión Technology | Infraestructura de hardware y software que soporta los datos. Sí, una empresa con datos y analistas puede tener baja madurez si carece de herramientas tecnológicas integradas para procesar o gobernar esa información eficientemente. | ¿Puede una organización con datos y analistas presentar madurez limitada? Justifique. |
| Dimensión Analytical Techniques | Conjunto de métodos matemáticos aplicados. La técnica descriptiva indica "qué pasó" (ej. reporte de ventas del mes), mientras que la diagnóstica explica "por qué pasó" (ej. correlacionar la caída de ventas con la falta de stock). | ¿Qué diferencia una técnica descriptiva de una diagnóstica? Proponga un ejemplo. |
| Organización que compite con analítica | Entidad que basa su estrategia y diferenciación central en los datos. No basta con tener tableros; requiere que la toma de decisiones basada en modelos estadísticos esté profundamente arraigada en su cultura operativa. | ¿Basta con disponer de tableros de control para competir con analítica? |
| Ventaja competitiva basada en datos | Posición de superioridad en el mercado lograda mediante *insights* exclusivos. Es difícil de imitar porque no depende solo de comprar tecnología, sino del volumen histórico de datos acumulados y una cultura organizacional madura. | ¿Por qué razones esta ventaja puede resultar difícil de imitar por un competidor? |
| Impacto de la IA generativa en la empresa | Transformación operativa basada en modelos que automatizan la síntesis y creación de información. Tareas como la decisión final de inversión o la validación directa de cálculos de riesgo no deben delegarse sin revisión humana experta. | ¿Qué tarea del análisis financiero no debería delegarse a un sistema generativo? |
| Perfil analítico comparado | Evaluación de métricas o niveles de madurez entre distintas entidades. Para ser comparables de forma justa, las empresas deben pertenecer a la misma industria, compartir modelos de negocio similares o usar estándares contables equivalentes. | ¿Qué condiciones deben cumplirse para que dos empresas sean comparables? |
| Indicador financiero como evidencia | Métrica extraída de estados contables usada para fundamentar el diagnóstico de una empresa. Pierde validez si se interpreta fuera de contexto, por ejemplo, omitiendo la estacionalidad del sector o distorsiones macroeconómicas coyunturales. | ¿En qué caso un indicador correctamente calculado no constituye evidencia válida? |

**Criterio de cierre.** No se avanza al desarrollo práctico mientras existan conceptos sin definición registrada.

## Marco conceptual — Del modelo DELTA al modelo DELTA Plus

El modelo DELTA fue formulado por Davenport, Harris y Morison en *Analytics at Work* (2010) como un instrumento para evaluar la madurez analítica de una organización a partir de cinco dimensiones: **Data**, referida a la disponibilidad, integración y calidad de los datos; **Enterprise**, que examina si la capacidad analítica se encuentra dispersa en áreas aisladas o coordinada a escala institucional; **Leadership**, que valora el compromiso efectivo de la dirección con la decisión basada en evidencia; **Targets**, que analiza si los esfuerzos analíticos se concentran en los objetivos de mayor valor para el negocio; y **Analysts**, que considera la disponibilidad de profesionales con las competencias requeridas. El principio central del modelo establece que estas dimensiones deben progresar de manera conjunta, dado que un avance aislado en una de ellas no eleva la madurez global de la organización.

Con posterioridad, y ante la transformación del entorno tecnológico de la analítica empresarial, el modelo fue ampliado bajo la denominación **DELTA Plus**, que incorpora dos dimensiones adicionales. La primera es **Technology**, que comprende la infraestructura y las plataformas que permiten desarrollar y desplegar capacidades analíticas: soluciones en la nube, *data warehouses*, *data lakes* y arquitecturas *lakehouse*, entornos de procesamiento de grandes volúmenes de información, herramientas de visualización, plataformas analíticas corporativas, componentes de código abierto e infraestructura para el desarrollo de modelos de aprendizaje automático e inteligencia artificial. Su incorporación obedece a una constatación de orden práctico: una organización puede disponer de datos pertinentes y de analistas calificados y presentar, aun así, una madurez analítica limitada si carece de la tecnología necesaria para almacenar, procesar, analizar y poner en uso esa información de manera periódica y reproducible.

La segunda dimensión añadida es **Analytical Techniques**, que evalúa la variedad y la sofisticación de las técnicas empleadas para convertir los datos en información útil para la decisión. Estas técnicas admiten una progresión reconocible: la analítica **descriptiva** establece qué ocurrió; la **diagnóstica** indaga por qué ocurrió; la **predictiva** estima qué podría ocurrir; y la **prescriptiva** determina qué conviene hacer. Sobre esa base se incorporan progresivamente el aprendizaje automático, las técnicas de segmentación, los modelos de clasificación, los pronósticos, los métodos de optimización y las aplicaciones de inteligencia artificial. Dichas técnicas se presentan aquí como ilustración de la evolución de la capacidad analítica de una organización y serán objeto de desarrollo en semanas posteriores del curso; el presente laboratorio se sitúa en los niveles descriptivo y diagnóstico.

### Tabla comparativa

| | Modelo DELTA (2010) | Modelo DELTA Plus |
|---|---|---|
| **Dimensiones** | Cinco | Siete |
| **Componentes** | Data, Enterprise, Leadership, Targets, Analysts | Las cinco anteriores, más Technology y Analytical Techniques |
| **Objeto de evaluación** | Condiciones organizacionales de la analítica | Condiciones organizacionales y capacidad técnica de ejecución |
| **Pregunta que responde** | ¿Dispone la organización de datos, dirección, objetivos y personas? | Además de lo anterior, ¿cuenta con la tecnología y las técnicas para ejecutarlo de manera sostenida? |
| **Limitación que corrige** | — | Una organización con datos y analistas puede presentar madurez limitada por ausencia de tecnología o por el uso de técnicas elementales |

> **Precisión conceptual.** DELTA y DELTA Plus no constituyen modelos alternativos. DELTA Plus es una ampliación del modelo original: conserva íntegramente sus cinco dimensiones y añade dos. Al emitir un diagnóstico debe indicarse explícitamente cuál de los dos marcos se está aplicando.


## Actividad 2 — Desarrollo práctico y ejecución

**Propósito:** descargar información financiera anual y real de empresas del sector azucarero peruano, construir un perfil comparado de rentabilidad, liquidez y endeudamiento, y relacionarlo prudentemente con el contexto sectorial del BCRP mediante pandas, Polars, DuckDB y Plotly.

**Empresas:** Casa Grande S.A.A., Cartavio S.A.A., Agroindustrial Laredo S.A.A. y Empresa Agroindustrial Pomalca S.A.A.


## Paso 1: Preparación del entorno


In [0]:
# Ejecute esta celda en Google Colab. Las versiones se fijan para favorecer la reproducibilidad.
%pip install -q polars==1.17.1 duckdb==1.1.3 econdata==1.0.17

import json, html, re, unicodedata
from xml.etree import ElementTree as ET

import requests
import pandas as pd
import numpy as np
import polars as pl
import duckdb
import plotly.graph_objects as go
from IPython.display import display
from econdata import BCRP

pd.set_option("display.width", 160)
pd.set_option("display.max_columns", 40)
print("Entorno listo.")


Note: you may need to restart the kernel using %restart_python or dbutils.library.restartPython() to use updated packages.
Entorno listo.


## Paso 2: Descarga de información financiera real desde la SMV

Se emplea el servicio web oficial de la SMV para consultar las principales cuentas y el Estado de Situación Financiera. Las operaciones utilizadas son `obtener_InfoFinanciera` y `obtener_BalanceGeneral`. El nombre oficial `obtener_GanciaPerdida` también se conserva para consultas del Estado de Resultados.


In [0]:
SERVICIO_SMV = "https://mvnet.smv.gob.pe/ws_od_eeff/WebServiceInfoFinanciera.asmx"
EJERCICIO = 2024
PERIODO = "A"   # Anual
TIPO = "I"      # Individual
AZUCARERAS = ["CASA GRANDE", "CARTAVIO", "AGROINDUSTRIAL LAREDO", "POMALCA"]

def etiqueta_local(etiqueta):
    '''Retira el espacio de nombres de una etiqueta XML.'''
    return str(etiqueta).split("}")[-1]

def payload_a_dataframe(payload):
    '''Convierte el contenido JSON o XML incluido en la respuesta SOAP en DataFrame.'''
    payload = html.unescape(str(payload)).strip()
    try:
        objeto = json.loads(payload)
        if isinstance(objeto, dict):
            objeto = objeto.get("rows", objeto.get("data", objeto))
        if isinstance(objeto, dict):
            objeto = [objeto]
        return pd.DataFrame(objeto)
    except json.JSONDecodeError:
        raiz = ET.fromstring(payload)
        registros = []
        for nodo in raiz.iter():
            hijos = list(nodo)
            if len(hijos) >= 5 and all(not list(hijo) for hijo in hijos):
                registros.append({etiqueta_local(h.tag): h.text for h in hijos})
        return pd.DataFrame(registros)

def descargar_smv(operacion, ejercicio=EJERCICIO, periodo=PERIODO, tipo=TIPO):
    '''Consulta una operación oficial del servicio SOAP de la SMV.'''
    sobre = f'''<?xml version="1.0" encoding="utf-8"?>
    <soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
      <soap:Body><{operacion} xmlns="http://tempuri.org/">
        <Ejercicio>{ejercicio}</Ejercicio><Periodo>{periodo}</Periodo><Tipo>{tipo}</Tipo>
      </{operacion}></soap:Body>
    </soap:Envelope>'''
    respuesta = requests.post(
        SERVICIO_SMV,
        data=sobre.encode("utf-8"),
        timeout=90,
        headers={
            "Content-Type": "text/xml; charset=utf-8",
            "SOAPAction": f'"http://tempuri.org/{operacion}"',
        },
    )
    respuesta.raise_for_status()
    raiz = ET.fromstring(respuesta.content)
    resultado = next(
        (n for n in raiz.iter() if etiqueta_local(n.tag) == f"{operacion}Result"),
        None,
    )
    if resultado is None or not (resultado.text or "").strip():
        raise ValueError(f"La SMV no devolvió datos para {operacion}.")
    datos = payload_a_dataframe(resultado.text)
    if datos.empty:
        raise ValueError(f"La respuesta de {operacion} no contiene registros.")
    return datos

principales = descargar_smv("obtener_InfoFinanciera")
balance = descargar_smv("obtener_BalanceGeneral")

print("Principales cuentas:", principales.shape)
print("Estado de situación financiera:", balance.shape)
display(principales.head(3))


Principales cuentas: (276, 16)
Estado de situación financiera: (20574, 15)


,RPJ,TipoEmpresa,TipoSector,NombreEmpresa,RUC,CIIU,Ejercicio,TipoInformacion,Trimestre,Moneda,MetodoFlujoEfectivo,ActivoTotal,PatrimonioTotal,TotalIngreso,UtilidadNeta,PasivoTotal
0,L00474,EMPRESAS MERCADO ALTERNATIVO DE VALORES,DIVERSOS,A. JAIME ROJAS REPRESENTACIONES GENERALES S.A.,20102032951,5190,2024,Anual Individual,Anual,Soles,M todo Directo,77710,39497,77196,6786,38213
1,I00004,SOCIEDADES ADMINISTRADORAS DE FONDOS DE INVERSION,,AC CAPITALES SOCIEDAD ADMINISTRADORA DE FONDOS...,20504893295,6430,2024,Anual Individual,Anual,Soles,M todo Directo,9070,7373,4880,-85,1697
2,OE7511,SOCIEDADES ADMINISTRADORAS DE FONDOS DE INVERSION,,ACRES SOCIEDAD ADMINISTRADORA DE FONDOS DE INV...,20601498996,6430,2024,Anual Individual,Anual,Soles,M todo Indirecto,2724,2238,2348,176,486


## Paso 3: Normalización, filtrado y validación del sector

Se normalizan los nombres de las columnas, se filtran las cuatro empresas y se verifica que todas estén representadas. No se utiliza una base simulada ni un respaldo inventado.


In [0]:
def nombre_columna(texto):
    texto = re.sub(r"(?<!^)(?=[A-Z])", "_", str(texto))
    texto = unicodedata.normalize("NFKD", texto).encode("ascii", "ignore").decode()
    return re.sub(r"[^a-z0-9]+", "_", texto.lower()).strip("_")

def preparar(df):
    df = df.copy()
    df.columns = [nombre_columna(c) for c in df.columns]
    return df

def resolver_columna(df, candidatos):
    '''Encuentra una columna aun cuando la SMV cambie mayúsculas o convenciones de nombre.'''
    disponibles = {nombre_columna(c): c for c in df.columns}
    for candidato in candidatos:
        clave = nombre_columna(candidato)
        if clave in disponibles:
            return disponibles[clave]
    raise KeyError(f"No se encontró ninguna de estas columnas: {candidatos}. Disponibles: {list(df.columns)}")

def estandarizar_empresa(df):
    df = preparar(df)
    origen = resolver_columna(df, ["nombre_empresa", "razon_social", "empresa"])
    return df.rename(columns={origen: "nombre_empresa"})

def filtrar_azucareras(df):
    patron = "|".join(map(re.escape, AZUCARERAS))
    return df[
        df["nombre_empresa"].astype(str).str.upper().str.contains(patron, regex=True, na=False)
    ].copy()

principales = filtrar_azucareras(estandarizar_empresa(principales))
balance = filtrar_azucareras(estandarizar_empresa(balance))

empresas = sorted(principales["nombre_empresa"].dropna().unique())
print("Empresas encontradas:")
for empresa in empresas:
    print("-", empresa)

faltantes = [nombre for nombre in AZUCARERAS if not any(nombre in e.upper() for e in empresas)]
if faltantes:
    raise ValueError(f"No se localizaron todas las empresas. Faltantes: {faltantes}")


Empresas encontradas:
- AGROINDUSTRIAL LAREDO S.A.A.
- CARTAVIO SOCIEDAD ANONIMA ABIERTA
- CASA GRANDE SOCIEDAD ANONIMA ABIERTA
- EMPRESA AGROINDUSTRIAL POMALCA S.A.A.


### Paso 3: La cuenta correcta no se busca por su nombre

El Estado de Situación Financiera trae 69 cuentas por empresa, y **más de una contiene el texto «Activos Corrientes»**, pero solo una es el total. Filtrar por la descripción es un error frecuente y silencioso: no lanza ninguna excepción, devuelve un número, y ese número está mal.

Ejecute la celda siguiente y observe el catálogo: la cuenta `1D0122` —«Activos Corrientes Distintos al Efectivo Pignorados como Garantía Colateral»— vale 0 y, de usarse, produce una razón corriente igual a cero.


In [0]:
# Catálogo de cuentas cuyo texto contiene "corriente", para una de las empresas.
muestra = balance[balance["nombre_empresa"] == empresas[0]]
columna_desc = resolver_columna(muestra, ["descripcion_cuenta", "descripcion"])
catalogo = muestra[
    muestra[columna_desc].astype(str).str.contains("orriente", na=False)
][["cuenta", columna_desc, "monto1"]]
print("Empresa de referencia:", empresas[0])
display(catalogo.reset_index(drop=True))

TOTAL_ACTIVO_CORRIENTE = "1D01ST"
TOTAL_PASIVO_CORRIENTE = "1D03ST"

print("Códigos declarados:", TOTAL_ACTIVO_CORRIENTE, TOTAL_PASIVO_CORRIENTE)


Empresa de referencia: AGROINDUSTRIAL LAREDO S.A.A.


,cuenta,descripcion_cuenta,monto1
0,1D0122,Activos Corrientes Distintos al Efectivo Pigno...,0.0
1,1D0118,Total Activos Corrientes Distintos de los Acti...,137966.0
2,1D0115,Activos no Corrientes o Grupos de Activos para...,0.0
3,1D01ST,Total Activos Corrientes,137966.0
4,1D0223,"Activos por Impuestos Corrientes, no Corrientes",0.0
5,1D0224,Activos no Corrientes Distintos al Efectivo Pi...,0.0
6,1D02ST,Total Activos No Corrientes,802358.0
7,1D0315,Total de Pasivos Corrientes Distintos de Pasiv...,125303.0
8,1D03ST,Total Pasivos Corrientes,125303.0
9,1D0412,"Pasivos por Impuestos Corrientes, no Corriente",0.0


Códigos declarados: 1D01ST 1D03ST


## Paso 4: Construcción reproducible de los indicadores financieros

| Familia | Indicador | Fórmula |
|---|---|---|
| Rentabilidad | Margen neto | Utilidad neta / Ingresos |
| Rentabilidad | ROA | Utilidad neta / Activo total |
| Rentabilidad | ROE | Utilidad neta / Patrimonio |
| Liquidez | Razón corriente | Activo corriente / Pasivo corriente |
| Endeudamiento | Endeudamiento | Pasivo total / Activo total |
| Endeudamiento | Apalancamiento | Pasivo total / Patrimonio |

La razón corriente se obtiene del Estado de Situación Financiera mediante las descripciones oficiales de activos y pasivos corrientes.


In [0]:
def renombrar_primera(df, destino, candidatos):
    if destino in df.columns:
        return df
    return df.rename(columns={resolver_columna(df, candidatos): destino})

# Estandarización de las cuentas principales que devuelve obtener_InfoFinanciera.
for destino, candidatos in {
    "ventas": ["total_ingreso", "total_ingresos", "ingresos", "ventas_netas"],
    "utilidad_neta": ["utilidad_neta", "ganancia_perdida_neta", "resultado_neto"],
    "activo_total": ["activo_total", "total_activo", "total_activos"],
    "pasivo_total": ["pasivo_total", "total_pasivo", "total_pasivos"],
    "patrimonio": ["patrimonio_total", "total_patrimonio", "patrimonio"],
}.items():
    principales = renombrar_primera(principales, destino, candidatos)

columnas_base = [
    "nombre_empresa", "ventas", "utilidad_neta", "activo_total",
    "pasivo_total", "patrimonio"
]
perfil = principales[columnas_base].drop_duplicates("nombre_empresa").copy()
perfil = perfil.rename(columns={"nombre_empresa": "empresa"})

def a_numero(valor):
    '''Convierte importes de la SMV aunque incluyan separadores, espacios o paréntesis.'''
    if pd.isna(valor):
        return np.nan
    if isinstance(valor, (int, float, np.number)):
        return float(valor)
    texto = str(valor).strip().replace("\u00a0", "").replace(" ", "")
    negativo = texto.startswith("(") and texto.endswith(")")
    texto = texto.strip("()")
    # La SMV normalmente usa punto decimal. Las comas se eliminan como separador de miles.
    texto = texto.replace(",", "")
    texto = re.sub(r"[^0-9.\-]", "", texto)
    if texto in {"", "-", "."}:
        return np.nan
    numero = float(texto)
    return -numero if negativo else numero

for columna in ["ventas", "utilidad_neta", "activo_total", "pasivo_total", "patrimonio"]:
    perfil[columna] = perfil[columna].map(a_numero)

def extraer_total_corriente(df, codigo, nombre):
    col_codigo = resolver_columna(df, ["cuenta", "codigo_cuenta", "codigo"])
    monto = resolver_columna(df, ["monto1", "monto", "importe", "valor"])
    datos = df[df[col_codigo].astype(str).str.strip() == codigo][
        ["nombre_empresa", monto]].copy()
    datos[monto] = datos[monto].map(a_numero)
    datos = datos.dropna().drop_duplicates("nombre_empresa")
    if len(datos) < len(AZUCARERAS):
        raise ValueError(
            f"La cuenta {codigo} no está disponible para las cuatro empresas "
            f"en el ejercicio {EJERCICIO}."
        )
    return datos.rename(columns={"nombre_empresa": "empresa", monto: nombre})

perfil = perfil.merge(
    extraer_total_corriente(balance, TOTAL_ACTIVO_CORRIENTE, "activo_corriente"),
    on="empresa", how="inner")
perfil = perfil.merge(
    extraer_total_corriente(balance, TOTAL_PASIVO_CORRIENTE, "pasivo_corriente"),
    on="empresa", how="inner")

if len(perfil) < 4:
    raise ValueError("No fue posible construir el perfil completo de las cuatro empresas.")

columnas_calculo = [
    "ventas", "utilidad_neta", "activo_total", "pasivo_total", "patrimonio",
    "activo_corriente", "pasivo_corriente"
]
faltantes = perfil[columnas_calculo].isna().sum()
if faltantes.any():
    raise ValueError(
        "Existen cuentas sin valor numérico: " +
        ", ".join(f"{c} ({n})" for c, n in faltantes.items() if n)
    )

if (perfil[["ventas", "activo_total", "patrimonio", "pasivo_corriente"]] == 0).any().any():
    raise ValueError("Existe un denominador igual a cero; revise las cuentas antes de calcular.")

perfil["margen_neto"] = perfil["utilidad_neta"] / perfil["ventas"]
perfil["roa"] = perfil["utilidad_neta"] / perfil["activo_total"]
perfil["roe"] = perfil["utilidad_neta"] / perfil["patrimonio"]
perfil["razon_corriente"] = perfil["activo_corriente"] / perfil["pasivo_corriente"]
perfil["endeudamiento"] = perfil["pasivo_total"] / perfil["activo_total"]
perfil["apalancamiento"] = perfil["pasivo_total"] / perfil["patrimonio"]

numeros = perfil.select_dtypes(include="number").to_numpy()
if np.isinf(numeros[~np.isnan(numeros)]).any():
    raise ValueError("Se obtuvo una división entre cero; revise las cuentas de la empresa afectada.")

display(perfil.round(3))


,empresa,ventas,utilidad_neta,activo_total,pasivo_total,patrimonio,activo_corriente,pasivo_corriente,margen_neto,roa,roe,razon_corriente,endeudamiento,apalancamiento
0,AGROINDUSTRIAL LAREDO S.A.A.,438934.0,55453.0,940324.0,327342.0,612982.0,137966.0,125303.0,0.126,0.059,0.090,1.101,0.348,0.534
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,535074.0,91846.0,926161.0,542687.0,383474.0,294933.0,307537.0,0.172,0.099,0.240,0.959,0.586,1.415
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,637681.0,73490.0,1923732.0,874083.0,1049649.0,429304.0,362479.0,0.115,0.038,0.070,1.184,0.454,0.833
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,234045.0,9636.0,1352469.0,679594.0,672875.0,168835.0,108618.0,0.041,0.007,0.014,1.554,0.502,1.010


**Pregunta 1.** Identifique la empresa con mayor ROE y la empresa con mayor endeudamiento. ¿Coinciden? Interprete la relación observada sin afirmar causalidad.

**Respuesta:** Con los datos completos de las cuatro empresas, **Cartavio** presenta el mayor ROE (24.0%) y también el mayor nivel de endeudamiento (58.6% del activo). Por lo tanto, **sí coinciden**.

La relación es aritmética, no causal: el ROE divide la utilidad neta entre el patrimonio. Al tener un mayor endeudamiento, el patrimonio de Cartavio es proporcionalmente menor frente a su pasivo (apalancamiento de 1.42 veces, contra 0.53 de Laredo o 0.83 de Casa Grande). Con una base patrimonial más pequeña, la utilidad produce un ROE mayor — no porque el endeudamiento "genere" rentabilidad, sino porque reduce el denominador sobre el que esa rentabilidad se mide.

El caso de **Pomalca** confirma la ausencia de causalidad: a pesar de tener un alto endeudamiento (50.2%), su ROE es el más bajo del grupo (1.4%), lo que demuestra que el apalancamiento financiero solo magnifica el indicador si la empresa genera suficiente utilidad neta. Un ROE alto apalancado en deuda no es evidencia, por sí solo, de mejor gestión operativa.

### Reto 1 — Predicción antes del ranking (5 minutos)

**Sin ordenar todavía la tabla**, responda en equipo:

1. ¿Qué empresa creen que presenta el mejor perfil financiero?
2. Sustenten la predicción con **dos indicadores** como mínimo.
3. ¿Qué indicador podría hacer cambiar su conclusión?

**Predicción del equipo:** Cartavio Sociedad Anónima Abierta.

**Justificación:** Presenta el mejor desempeño en eficiencia y uso de activos, evidenciado por el mayor Margen Neto (17.2%) y el mayor ROA (9.9%). El indicador que podría hacer cambiar esta conclusión es la Razón Corriente, ya que Cartavio es la única empresa con un valor inferior a 1 (0.96), lo que señala un potencial riesgo de liquidez a corto plazo frente a alternativas más sólidas en este aspecto como Laredo (1.10) o Casa Grande (1.18).

Después de ejecutar el ranking del Paso 5, vuelva a esta respuesta y escriba: **¿se confirmó o no la predicción y por qué?**

**Sí, se ha confirmado.** Cartavio domina el ranking superior en todas las métricas de rentabilidad (Margen Neto, ROA y ROE). El modelo de datos ratifica que asume mayor riesgo para lograr esa rentabilidad (es la empresa con mayor endeudamiento y menor liquidez), pero su capacidad de generar utilidades a partir de sus activos y ventas la posiciona con el mejor perfil financiero del grupo.

## Paso 5: Comparación sectorial con Polars y DuckDB


In [0]:
perfil_pl = pl.from_pandas(perfil)

ranking_roe = (
    perfil_pl
    .select([
        "empresa", "margen_neto", "roa", "roe", "razon_corriente",
        "endeudamiento", "apalancamiento"
    ])
    .sort("roe", descending=True)
)
ranking_roe


empresa,margen_neto,roa,roe,razon_corriente,endeudamiento,apalancamiento
str,f64,f64,f64,f64,f64,f64
"""CARTAVIO SOCIEDAD ANONIMA ABIE…",0.171651,0.099169,0.23951,0.959016,0.585953,1.415186
"""AGROINDUSTRIAL LAREDO S.A.A.""",0.126336,0.058972,0.090464,1.101059,0.348116,0.534016
"""CASA GRANDE SOCIEDAD ANONIMA A…",0.115246,0.038202,0.070014,1.184356,0.454368,0.832738
"""EMPRESA AGROINDUSTRIAL POMALCA…",0.041172,0.007125,0.014321,1.554392,0.502484,1.009986


In [0]:

CONSULTA = """
SELECT empresa,
       ROUND(margen_neto * 100, 1)    AS margen_neto_pct,
       ROUND(roa * 100, 1)            AS roa_pct,
       ROUND(roe * 100, 1)            AS roe_pct,
       ROUND(razon_corriente, 2)      AS razon_corriente,
       ROUND(endeudamiento * 100, 1)  AS endeudamiento_pct,
       ROUND(apalancamiento, 2)       AS apalancamiento
FROM perfil
ORDER BY roe_pct DESC
"""

consulta = duckdb.sql(CONSULTA).df()
display(consulta)

# Comprobación automática: DuckDB y Polars deben coincidir.
if list(consulta["empresa"]) == list(ranking_roe["empresa"]):
    print("Correcto: DuckDB y Polars devuelven el mismo ordenamiento.")
else:
    print("Los ordenamientos no coinciden. Revise las columnas y el ORDER BY.")


,empresa,margen_neto_pct,roa_pct,roe_pct,razon_corriente,endeudamiento_pct,apalancamiento
0,CARTAVIO SOCIEDAD ANONIMA ABIERTA,17.2,9.9,24.0,0.96,58.6,1.42
1,AGROINDUSTRIAL LAREDO S.A.A.,12.6,5.9,9.0,1.10,34.8,0.53
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,11.5,3.8,7.0,1.18,45.4,0.83
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,4.1,0.7,1.4,1.55,50.2,1.01


Correcto: DuckDB y Polars devuelven el mismo ordenamiento.


In [0]:
# Rankings complementarios para comparar dimensiones distintas del perfil.
ranking_comparado = pd.DataFrame({
    "empresa": perfil["empresa"],
    "rank_roe": perfil["roe"].rank(ascending=False, method="min").astype(int),
    "rank_liquidez": perfil["razon_corriente"].rank(ascending=False, method="min").astype(int),
    # En endeudamiento, un valor menor recibe mejor posición solo como criterio descriptivo de exposición financiera.
    "rank_endeudamiento": perfil["endeudamiento"].rank(ascending=True, method="min").astype(int),
})
ranking_comparado["cambio_max_posicion"] = (
    ranking_comparado[["rank_roe", "rank_liquidez", "rank_endeudamiento"]].max(axis=1)
    - ranking_comparado[["rank_roe", "rank_liquidez", "rank_endeudamiento"]].min(axis=1)
)
display(ranking_comparado.sort_values(["rank_roe", "empresa"]))


,empresa,rank_roe,rank_liquidez,rank_endeudamiento,cambio_max_posicion
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,1,4,4,3
0,AGROINDUSTRIAL LAREDO S.A.A.,2,3,1,2
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,3,2,2,1
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,4,1,3,3


### Reto 2 — Evidencia, inferencia y conclusión (5 minutos)

Clasifique cada afirmación como **Sustentada**, **No necesariamente** o **No puede concluirse** y justifique con cifras:

| Afirmación | Clasificación | Justificación con evidencia |
|---|---|---|
| La empresa con mayor ROE es necesariamente la más competitiva. | No necesariamente | Cartavio tiene el mayor ROE (24.0%) pero también la razón corriente más baja de las cuatro empresas (0.96, el único por debajo de 1) y el mayor endeudamiento (58.6%). Un ROE alto sostenido con liquidez ajustada y alto apalancamiento no equivale automáticamente a "más competitiva" — refleja un perfil de mayor riesgo. |
| Una empresa con mayor endeudamiento posee necesariamente menor madurez analítica. | No puede concluirse | El endeudamiento es una decisión de estructura de capital (cuánta deuda vs. patrimonio financia la operación). La madurez analítica se refiere a procesos, cultura de datos y capacidad tecnológica — ninguna cuenta del Estado de Situación Financiera mide esto. No hay ninguna variable en el dataset que conecte de forma causal ambos conceptos. |
| Obtener un ordenamiento consistente con Polars y DuckDB fortalece la reproducibilidad del análisis. | Sustentada | La celda de comprobación automática compara explícitamente si ambos motores devuelven el mismo orden de empresas. Que dos motores independientes, operando sobre los mismos datos, produzcan exactamente el mismo resultado es la definición operativa de reproducibilidad técnica. |

**Importante:** reproducibilidad no significa causalidad ni convierte un indicador financiero en evidencia directa de madurez organizacional.

**Pregunta 2.** Compare los rankings por rentabilidad, liquidez y endeudamiento. Explique qué revela el cambio de posición sobre el perfil competitivo de cada empresa.

**Respuesta:** Con Casa Grande y Cartavio (2 de 4 empresas confirmadas), el ranking cambia de forma opuesta según el criterio:

- **Por ROE:** 1.º Cartavio (23.95%), 2.º Casa Grande (7.00%).
- **Por liquidez (razón corriente):** 1.º Casa Grande (1.18), 2.º Cartavio (0.96).
- **Por endeudamiento (menor es mejor posición):** 1.º Casa Grande (45.44%), 2.º Cartavio (58.60%).

Cartavio lidera en rentabilidad pero queda último en liquidez y en exposición al endeudamiento; Casa Grande muestra el patrón inverso: menor rentabilidad, pero estructura financiera más conservadora y liquidez por encima de 1. El cambio de posición revela que **ninguna de las dos empresas domina en las tres dimensiones a la vez** — el perfil competitivo depende de qué riesgo se esté dispuesto a asumir: Cartavio ofrece más retorno con más riesgo de corto plazo, Casa Grande ofrece más solidez con menos retorno.

**Pendiente:** con solo 2 empresas el "cambio de posición máximo" (`cambio_max_posicion`) solo puede ser 0 o 1 — la métrica se vuelve informativa recién con las 4 empresas. Repetir esta lectura cuando Laredo y Pomalca estén disponibles.

---


## Paso 6: Visualización del perfil comparado


In [0]:
indicadores_pct = ["margen_neto", "roa", "roe", "endeudamiento"]
nombres_indicadores = {
    "margen_neto": "Margen neto",
    "roa": "ROA",
    "roe": "ROE",
    "endeudamiento": "Endeudamiento",
}
perfil_grafico = perfil.copy()
perfil_grafico["empresa_corta"] = (
    perfil_grafico["empresa"].str.upper()
    .map(lambda x: "Laredo" if "LAREDO" in x else
                   "Cartavio" if "CARTAVIO" in x else
                   "Casa Grande" if "CASA GRANDE" in x else
                   "Pomalca" if "POMALCA" in x else x.title())
)
fig = go.Figure()
for indicador in indicadores_pct:
    fig.add_trace(go.Bar(
        name=nombres_indicadores[indicador],
        x=perfil_grafico["empresa_corta"],
        y=perfil_grafico[indicador] * 100,
        text=(perfil_grafico[indicador] * 100).round(1),
        texttemplate="%{text:.1f}%",
        textposition="outside",
        customdata=perfil_grafico[["empresa"]],
        hovertemplate="%{customdata[0]}<br>%{fullData.name}: %{y:.2f}%<extra></extra>",
    ))
fig.update_layout(
    title=f"Perfil financiero comparado — sector azucarero ({EJERCICIO})",
    barmode="group",
    yaxis_title="Porcentaje (%)",
    height=540,
    width=1050,
    template="plotly_white",
    margin=dict(t=110, b=90, l=80, r=40),
    legend=dict(orientation="h", yanchor="bottom", y=1.02, xanchor="left", x=0),
)
fig.show()

fig_liquidez = go.Figure(go.Bar(
    x=perfil_grafico["empresa_corta"],
    y=perfil_grafico["razon_corriente"],
    marker_color="#00A9CE",
    text=perfil_grafico["razon_corriente"].round(2),
    texttemplate="%{text:.2f}",
    textposition="outside",
    customdata=perfil_grafico[["empresa"]],
    hovertemplate="%{customdata[0]}<br>Razón corriente: %{y:.2f}<extra></extra>",
))
fig_liquidez.update_layout(
    title="Razón corriente por empresa",
    yaxis_title="Veces",
    height=460,
    width=900,
    template="plotly_white",
    margin=dict(t=80, b=80, l=80, r=40),
)
fig_liquidez.update_yaxes(rangemode="tozero")
fig_liquidez.show()


## Paso 7: Contexto sectorial con datos específicos del BCRP

Se emplean series oficiales sobre tipo de cambio, producción de caña y exportaciones de azúcar. Estas series describen el entorno; por sí solas no demuestran la causa de las diferencias financieras entre empresas.


In [0]:
SERIES = {
    "PN01209PM": "tipo_cambio",
    "PN01790AM": "produccion_cana_azucar_miles_t",
    "PN38774BM": "export_azucar_fob_millones_usd",
    "PN38775BM": "volumen_exportado_azucar",
}

contexto = BCRP.get_data(SERIES, "2020-01", "2024-12")
print("Datos del BCRP:", contexto.shape)
display(contexto.tail(12))


Datos del BCRP: (60, 4)


,tipo_cambio,produccion_cana_azucar_miles_t,export_azucar_fob_millones_usd,volumen_exportado_azucar
2024-01-01,3.743667,755.602441,7.547009,10.321329
2024-02-01,3.830905,814.525062,4.398561,5.993531
2024-03-01,3.712947,563.626363,1.911468,2.355260
2024-04-01,3.718409,634.675782,3.839971,4.951844
2024-05-01,3.735273,812.740966,3.235732,3.843619
2024-06-01,3.789526,830.637431,0.481814,0.618096
2024-07-01,3.769500,840.996455,0.919740,1.246593
2024-08-01,3.745550,1017.287110,1.010892,1.358606
2024-09-01,3.772143,930.472111,3.097093,4.088378
2024-10-01,3.757500,1053.902968,1.906427,2.543543


In [0]:

contexto_grafico = contexto.copy()
columnas_numericas = contexto_grafico.select_dtypes(include="number").columns
base = contexto_grafico[columnas_numericas].apply(lambda s: s.dropna().iloc[0])
indice_100 = contexto_grafico[columnas_numericas].divide(base).multiply(100)

fig_contexto = go.Figure()
for columna in indice_100.columns:
    fig_contexto.add_trace(go.Scatter(
        x=indice_100.index,
        y=indice_100[columna],
        mode="lines",
        name=columna,
    ))
fig_contexto.update_layout(
    title="Contexto del sector azucarero — índice base 100",
    yaxis_title="Índice",
    xaxis_title="Periodo",
    height=480,
)
fig_contexto.show()


**Pregunta 3.** Describa la evolución del tipo de cambio, la producción de caña y las exportaciones de azúcar. Relacione ese contexto con el perfil financiero y diferencie expresamente:

*   **Evidencia:** Durante 2024, el tipo de cambio se mantuvo en una banda estrecha y relativamente estable, oscilando entre S/ 3.71 (marzo) y S/ 3.83 (febrero). La producción de caña de azúcar presentó una fuerte estacionalidad: tocó su punto más bajo en marzo (563.6 miles de toneladas) y alcanzó su pico máximo en octubre (1,053.9 miles de toneladas). Las exportaciones fueron sumamente volátiles, iniciando el año con fuerza en enero ($7.54 millones FOB) pero desplomándose drásticamente hacia mediados de año, con un mínimo en junio ($0.48 millones FOB).
*   **Inferencia:** La estabilidad cambiaria sugiere que las posiciones de rentabilidad (como el ROE de 24.0% de Cartavio) no son producto de ganancias extraordinarias por diferencia de cambio, sino de su estructura operativa. Asimismo, la drástica estacionalidad de la producción (baja en el primer trimestre, alta en el tercero y cuarto) obliga a las empresas a gestionar ciclos de caja muy marcados. Esto explica por qué empresas como Cartavio mantienen una razón corriente baja (0.96) y alto endeudamiento: probablemente recurren a deuda de corto plazo para sostener la operación durante los meses de baja producción y menores exportaciones.
*   **Opinión:** Considero que el perfil conservador de Laredo o Casa Grande es más adecuado para las condiciones de este sector. Mantener un endeudamiento superior al 58% (como Cartavio) en una industria donde las exportaciones pueden caer abruptamente de un mes a otro (como ocurrió entre enero y junio) asume un riesgo de liquidez excesivo, priorizando el retorno para el accionista sobre la resiliencia operativa frente a los choques del mercado agroindustrial.

### Reto 3 — Diagnóstico provisional de madurez y DELTA (8 minutos)

Seleccione **una** de las cuatro empresas. No se busca “adivinar” su madurez: se busca reconocer qué puede y qué no puede sostenerse con la evidencia disponible.

| Aspecto | Respuesta del equipo |
|---|---|
| Empresa seleccionada | Casa Grande S.A.A. |
| Evidencia financiera observada | ROE 7.00%, endeudamiento 45.44%, razón corriente 1.18 (2024) — perfil financiero conservador: rentabilidad moderada, liquidez positiva, menor dependencia de deuda que Cartavio. |
| Estadio de evolución analítica compatible **de manera provisional** | Descriptivo/diagnóstico bajo — los datos permiten ver *qué* pasó (los ratios), pero nada en los estados financieros indica si la empresa usa esos datos de forma sistemática para decidir. |
| Dimensión DELTA evaluable parcialmente | *Data*: se puede evaluar si existen datos financieros confiables y auditados (sí los hay, y son públicos). |
| Dimensiones DELTA que no pueden verificarse | *Enterprise*, *Leadership*, *Targets*, *Analysts* — ninguna tiene una variable observable en un estado financiero. |
| Evidencia organizacional adicional requerida | Entrevistas o documentación interna sobre uso de BI/analítica, estructura del equipo de datos, y objetivos estratégicos declarados por la gerencia. |

**Cierre del reto:** la **evidencia** es el ratio calculado directamente de la cifra auditada (ej. razón corriente = 1.18); la **inferencia** es la lectura razonada sobre esa cifra (ej. "la empresa puede cubrir sus obligaciones de corto plazo sin apremio"); la **opinión** es cualquier juicio de valor no verificable con estos datos (ej. "Casa Grande está mejor gestionada que Cartavio") — las tres se parecen en el texto, pero solo la primera es un hecho.

---


### Pausa metodológica — ¿qué NO podemos afirmar?

Los estados financieros permiten comparar resultados y estructura financiera, pero **no prueban por sí solos** liderazgo analítico, cultura de datos, disponibilidad de analistas, integración empresarial de la analítica ni uso de IA generativa. Por ello, cualquier estadio o perfil DELTA derivado de este laboratorio debe presentarse como **hipótesis provisional**, no como diagnóstico organizacional definitivo.


## Actividad 3 — Informe de perfil analítico comparado

Responda en términos de negocio y sustente cada afirmación con las cifras obtenidas.

**A.** Interprete tres hallazgos del perfil comparado y explique, para cada uno, su implicancia competitiva en el sector.

**Respuesta:**
1. **Cartavio lidera en rentabilidad pero asume el mayor riesgo, mientras Laredo equilibra.** Cartavio obtiene el mayor ROE (24.0%) con el mayor endeudamiento (58.6%). Laredo logra el segundo mejor ROE (9.0%) operando con el menor endeudamiento del sector (34.8%). Implicancia: Laredo prioriza la solidez patrimonial y un crecimiento conservador, mientras que Cartavio utiliza agresivamente la deuda para maximizar el retorno al accionista.
2. **Pomalca tiene exceso de liquidez sin generar retornos.** Pomalca presenta la mayor razón corriente (1.55) pero el ROE más bajo del grupo (1.4%). Implicancia: Una alta liquidez no es una ventaja si el capital está inmovilizado. Pomalca es ineficiente en la asignación de sus recursos a corto plazo, perdiendo costo de oportunidad frente a competidores. 
3. **La liquidez de Cartavio (0.96) está por debajo de 1.** Es la única de las cuatro empresas en esta situación. Implicancia: En el corto plazo, Cartavio depende críticamente de que sus activos corrientes roten a tiempo para cubrir sus obligaciones inmediatas. Es un punto de vulnerabilidad operativa ante cualquier choque en las ventas o producción.

**B.** Determine si los estados financieros permiten ubicar de manera concluyente a cada empresa en uno de los cinco estadios de evolución analítica. Formule, cuando corresponda, una **clasificación provisional** y precise qué evidencia organizacional adicional se necesita para confirmarla.

**Respuesta:** No, de manera concluyente no. Los estados financieros describen resultados y estructura financiera, no procesos internos de decisión. Como máximo, permiten una clasificación **provisional**: las cuatro empresas muestran indicadores consistentes con una etapa "descriptiva" de uso de datos financieros (se miden y reportan resultados con rigor, cumpliendo estándares de auditoría), pero no hay evidencia de estadios más avanzados (analítica predictiva o prescriptiva). Confirmar cualquier estadio real requeriría evidencia organizacional adicional: qué herramientas de BI usan, si tienen un equipo de datos dedicado, y si las decisiones gerenciales se basan en modelos analíticos formales.

**C.** A partir del modelo DELTA, ¿qué dimensión resulta más difícil de evaluar únicamente con estados financieros? Explique por qué y qué información adicional se requeriría.

**Respuesta:** *Analysts* (A) es la más difícil: ningún estado financiero reporta si la empresa cuenta con personal capaz de analizar datos, ni su nivel de especialización. A diferencia de *Data*, que sí se puede inferir parcialmente (la existencia de estados auditados y trazables es en sí misma un indicador de disciplina de datos), *Analysts* requiere información puramente organizacional: organigramas, perfiles de puestos o entrevistas directas con el liderazgo de la empresa.

**D.** Formule dos recomendaciones sustentadas exclusivamente en los datos observados, dirigidas a la empresa con el perfil competitivo más débil.

**Respuesta:** Con los datos completos, **Pomalca** muestra el perfil competitivo más débil del sector (ROE de 1.4%, margen neto de solo 4.1%, pero endeudamiento del 50.2%). Dos recomendaciones basadas en estas cifras:
1. **Reasignar el exceso de capital de trabajo:** Pomalca tiene la razón corriente más alta (1.55). Debe evaluar el uso de esos activos corrientes inmovilizados para amortizar parte de su deuda, lo que reduciría gastos financieros y mejoraría su deficiente rentabilidad neta.
2. **Ejecutar una auditoría estricta de eficiencia de costos:** Con un margen neto de apenas 4.1% (frente al 17.2% de Cartavio o 12.6% de Laredo), la empresa es altamente ineficiente convirtiendo ventas en utilidad. Necesita identificar qué rubros operativos están erosionando sus márgenes.

**E.** Explique cómo la inteligencia artificial generativa podría modificar el análisis de competitividad. Distinga entre usos plausibles y afirmaciones no verificables.

**Respuesta:** Usos plausibles: automatizar la extracción y normalización de cuentas contables desde los reportes en PDF de la SMV, generar primeros borradores de interpretación de ratios financieros para revisión humana, y detectar anomalías en series extensas de estados financieros más rápido que una auditoría manual. Afirmaciones no verificables: que un modelo de IA generativa pueda "predecir" el desempeño futuro de una empresa o "diagnosticar" la madurez analítica interna de una organización sin acceso a sus datos operativos y cultura interna. Ambas cosas exceden lo que un LLM puede sostener sin incurrir en alucinaciones.

### Matriz de síntesis para la decisión

Complete esta matriz antes de redactar las conclusiones finales:

| Empresa | Fortaleza principal | Riesgo/debilidad principal | Evidencia numérica | Implicancia competitiva |
|---|---|---|---|---|
| Casa Grande | Estructura financiera conservadora, liquidez positiva | Rentabilidad intermedia/baja frente a los líderes | ROE 7.0%, razón corriente 1.18, endeudamiento 45.4% | Perfil de menor riesgo, pero menor retorno para el accionista |
| Cartavio | Mayor eficiencia en el uso de activos y mayor rentabilidad | Liquidez de corto plazo ajustada (por debajo de 1) y alto nivel de deuda | ROE 24.0%, razón corriente 0.96, endeudamiento 58.6% | Mayor retorno, pero con más exposición a riesgo financiero a corto plazo |
| Agroindustrial Laredo | Solidez patrimonial con el menor nivel de endeudamiento del sector | Rentabilidad inferior a la del líder (Cartavio) | Endeudamiento 34.8%, ROE 9.0%, razón corriente 1.10 | Perfil altamente resiliente y defensivo ante choques de mercado |
| Pomalca | Exceso de liquidez a corto plazo | Extrema ineficiencia para generar retornos y alto endeudamiento | ROE 1.4%, razón corriente 1.55, endeudamiento 50.2% | Destrucción potencial de valor; no justifica el riesgo asumido con la deuda |

Luego responda: **¿qué empresa requiere mayor atención gerencial según los indicadores observados y qué dos acciones recomendaría sin exceder la evidencia disponible?**

**Respuesta:** Con los datos de las cuatro empresas completos, **Pomalca** es la que requiere mayor atención gerencial urgente. Si bien la falta de liquidez de Cartavio era una alerta válida, Pomalca presenta un cuadro crítico de ineficiencia estructural: opera con un alto endeudamiento (50.2%) pero genera un ROE marginal (1.4%), manteniendo a la vez un exceso de liquidez ociosa (razón corriente de 1.55). 

Dos acciones dentro de lo que la evidencia permite sostener: 
(1) Evaluar la reasignación de sus activos corrientes inmovilizados (ese exceso sobre el nivel de 1.0) para amortizar parte de su deuda, reduciendo el gasto financiero. 
(2) Ejecutar una auditoría estricta de eficiencia sobre su estructura de costos operativos para identificar por qué su margen y rentabilidad son tan bajos comparados con el resto de la industria.

---

## Reto de aplicación y retroalimentación

En esta sección se aplicarán los procedimientos desarrollados durante la sesión a nuevas situaciones de análisis. Cada ejercicio requiere modificar, completar o construir código a partir de los datos financieros previamente procesados. Posteriormente, los resultados obtenidos deberán interpretarse brevemente desde una perspectiva empresarial. El propósito es comprobar la comprensión de las técnicas utilizadas y fortalecer la capacidad de adaptar el análisis ante nuevas preguntas de negocio.

**Indicaciones generales.** Los cinco ejercicios operan sobre `perfil_reto`, copia de trabajo del perfil financiero construido en la Actividad 2. Se emplea una copia para que ninguna modificación introducida en esta sección altere los resultados obtenidos con anterioridad. Los datos proceden íntegramente de la SMV; no corresponde generar ni sustituir valores en ningún caso. Cada respuesta escrita no debe exceder cuatro líneas.

**Duración en sesión:** 25 minutos. Los ejercicios que no concluyan durante la sesión se completan como avance del entregable y se depositan en el repositorio del equipo junto con el notebook ejecutado.


In [0]:
# Copia de trabajo para la sección de retos.
perfil_reto = perfil.copy()
print("Copia de trabajo creada:", perfil_reto.shape)
display(perfil_reto[["empresa", "ventas", "activo_total", "margen_neto", "roa", "roe"]].round(3))


Copia de trabajo creada: (4, 14)


,empresa,ventas,activo_total,margen_neto,roa,roe
0,AGROINDUSTRIAL LAREDO S.A.A.,438934.0,940324.0,0.126,0.059,0.090
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,535074.0,926161.0,0.172,0.099,0.240
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,637681.0,1923732.0,0.115,0.038,0.070
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,234045.0,1352469.0,0.041,0.007,0.014


### Ejercicio 1 — Recuperación de una nueva cuenta financiera del catálogo oficial

Durante la Actividad 2 se identificaron dos cuentas del Estado de Situación Financiera a partir de su **código oficial** y no de su descripción textual. Ese procedimiento constituye el criterio de trazabilidad del laboratorio, dado que el código permanece estable entre ejercicios mientras que la descripción admite variantes que conducen a errores silenciosos.

En este ejercicio deberá replicar el mismo procedimiento sobre una cuenta que aún no ha sido utilizada: el **Total de Activos No Corrientes**. Esta cuenta reúne los recursos de largo plazo de la empresa —terrenos, plantaciones, maquinaria e instalaciones industriales— y su participación sobre el activo total describe la estructura de inversión de cada empresa azucarera.

**Se solicita:**

1. Localizar en el catálogo de la SMV la cuenta correspondiente al Total de Activos No Corrientes e identificar su código oficial.
2. Recuperar el valor de esa cuenta para las cuatro empresas analizadas.
3. Incorporar la variable a `perfil_reto`.
4. Calcular el porcentaje que dicha cuenta representa respecto del activo total.

**Tiempo estimado:** 10 a 12 minutos.

In [0]:
TOTAL_ACTIVO_NO_CORRIENTE = "1D02ST"  # Código oficial de la SMV

perfil_reto = perfil.copy()

perfil_reto = perfil_reto.merge(
    extraer_total_corriente(balance, TOTAL_ACTIVO_NO_CORRIENTE, "activo_no_corriente"),
    on="empresa", how="inner"
)

perfil_reto["pct_activo_no_corriente"] = perfil_reto["activo_no_corriente"] / perfil_reto["activo_total"]

display(perfil_reto[["empresa", "activo_total", "activo_no_corriente", "pct_activo_no_corriente"]].round(3))

# Cross-check manual con las cifras oficiales ya confirmadas (Casa Grande y Cartavio):
verificacion_manual = pd.DataFrame({
    "empresa": ["CASA GRANDE S.A.A.", "CARTAVIO S.A.A."],
    "activo_no_corriente_esperado": [1_494_428, 631_228],
    "pct_esperado": [0.777, 0.682],
})

,empresa,activo_total,activo_no_corriente,pct_activo_no_corriente
0,AGROINDUSTRIAL LAREDO S.A.A.,940324.0,802358.0,0.853
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,926161.0,631228.0,0.682
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,1923732.0,1494428.0,0.777
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,1352469.0,1183634.0,0.875


**Pregunta 4.** ¿Qué empresa presenta la mayor proporción de activos no corrientes y qué podría indicar esto sobre su estructura de inversión?

**Respuesta:** Con la verificación manual disponible, Casa Grande tendría la mayor proporción (77.7% del activo total en activos no corrientes, frente a 68.2% de Cartavio) — pendiente de confirmar con el código oficial del catálogo. Una proporción más alta sugiere una base de inversión más concentrada en activos de largo plazo (tierras, plantaciones, maquinaria), consistente con su menor rotación de activos: más capital inmovilizado en infraestructura, menos velocidad para convertir esos activos en ventas.

---


### Ejercicio 2 — Construcción de un indicador de eficiencia en el uso de los activos

Los indicadores calculados en la Actividad 2 describen rentabilidad, liquidez y endeudamiento, pero ninguno mide la eficiencia con que la empresa emplea su base de activos para generar ventas. La rotación de activos cubre esa dimensión: expresa cuántas unidades monetarias de venta produce cada unidad monetaria invertida en activos, y se obtiene dividiendo las ventas entre el activo total.

En el sector azucarero este indicador resulta particularmente informativo, por tratarse de empresas intensivas en capital, con extensas superficies de cultivo e instalaciones industriales. Dos empresas pueden alcanzar rentabilidades semejantes mediante estrategias opuestas: una sostenida en márgenes elevados y otra en un mayor volumen de operación sobre la misma base de activos.

**Se solicita:**

1. Calcular la rotación de activos e incorporarla a `perfil_reto` en la columna `rotacion_activos`.
2. Generar un ranking de las empresas ordenado de mayor a menor según dicho indicador.
3. Comparar ese ordenamiento con el obtenido a partir del ROA calculado en la Actividad 2.

**Tiempo estimado:** 10 a 12 minutos.


In [0]:
perfil_reto["rotacion_activos"] = perfil_reto["ventas"] / perfil_reto["activo_total"]

ranking_rotacion = perfil_reto[["empresa", "rotacion_activos", "roa"]].sort_values(
    "rotacion_activos", ascending=False
)
display(ranking_rotacion.round(3))

,empresa,rotacion_activos,roa
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.578,0.099
0,AGROINDUSTRIAL LAREDO S.A.A.,0.467,0.059
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.331,0.038
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.173,0.007


**Pregunta 5.** ¿La empresa con mayor rotación de activos presenta también el mayor ROA? Explique brevemente a partir de los resultados obtenidos.

**Respuesta:** Sí, en las dos empresas confirmadas: Cartavio tiene la mayor rotación de activos (0.58) y también el mayor ROA (9.92%), frente a Casa Grande (rotación 0.33, ROA 3.82%). Ambos rankings coinciden, lo que sugiere que la ventaja de Cartavio en rentabilidad sobre activos proviene, al menos en parte, de usar sus activos con mayor intensidad para generar ventas — no solo de un margen más alto. Pendiente confirmar si el patrón se sostiene con las 4 empresas.

---


### Ejercicio 3 — Verificación numérica de la descomposición DuPont del ROA

La descomposición DuPont establece que la rentabilidad sobre activos puede expresarse como el producto de dos componentes: el margen neto, que mide la rentabilidad obtenida por cada unidad vendida, y la rotación de activos, que mide la intensidad con que se emplea la base de activos.

El valor analítico de esta descomposición no reside en la identidad algebraica, sino en la explicación que aporta: permite establecer si una rentabilidad elevada proviene del precio y del control de costos, del volumen de operación, o de la concurrencia de ambos factores. Corresponde comprobar numéricamente la identidad sobre los datos reales de la SMV, dado que toda relación teórica empleada como evidencia debe verificarse antes de sustentar una conclusión.

**Se solicita:**

1. Construir la columna `roa_dupont` como el producto del margen neto por la rotación de activos.
2. Construir la columna `diferencia_roa` como la diferencia entre el ROA obtenido en la Actividad 2 y `roa_dupont`.
3. Verificar numéricamente la magnitud de esa diferencia y determinar si resulta admisible, considerando el redondeo de los importes reportados por la SMV.

**Tiempo estimado:** 10 a 12 minutos.


In [0]:
perfil_reto["roa_dupont"] = perfil_reto["margen_neto"] * perfil_reto["rotacion_activos"]
perfil_reto["diferencia_roa"] = perfil_reto["roa"] - perfil_reto["roa_dupont"]

display(perfil_reto[["empresa", "margen_neto", "rotacion_activos", "roa", "roa_dupont", "diferencia_roa"]].round(5))


,empresa,margen_neto,rotacion_activos,roa,roa_dupont,diferencia_roa
0,AGROINDUSTRIAL LAREDO S.A.A.,0.12634,0.46679,0.05897,0.05897,-0.0
1,CARTAVIO SOCIEDAD ANONIMA ABIERTA,0.17165,0.57773,0.09917,0.09917,0.0
2,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,0.11525,0.33148,0.03820,0.03820,0.0
3,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,0.04117,0.17305,0.00712,0.00712,0.0


**Pregunta 6.** ¿Qué componente parece explicar mejor el ROA de la empresa líder: el margen neto, la rotación de activos o ambos?

**Respuesta:** Cartavio es la empresa líder en ROA entre las confirmadas. Su margen neto (17.17%) también es más alto que el de Casa Grande (11.53%), pero la brecha en rotación de activos es proporcionalmente mayor (0.58 vs. 0.33, casi el doble) que la brecha en margen (1.5 veces). Esto sugiere que **ambos** componentes contribuyen, pero la rotación de activos pesa algo más en la diferencia de ROA entre estas dos empresas.

---


### Ejercicio 4 — Consulta empresarial con condiciones múltiples en DuckDB

En la Actividad 2 se empleó DuckDB para reproducir un ordenamiento ya obtenido con Polars, con la finalidad de comprobar la reproducibilidad del resultado entre dos motores distintos. Una consulta de negocio, sin embargo, no se limita a ordenar: delimita un subconjunto de la población mediante criterios definidos previamente por la gerencia.

En este ejercicio deberá construir una consulta que identifique aquellas empresas que satisfacen **simultáneamente** dos condiciones financieras —capacidad de atender obligaciones de corto plazo y generación de rentabilidad positiva para el accionista— y que presente el resultado ordenado según su exposición al endeudamiento. Este es el procedimiento habitual mediante el cual un área de riesgos delimita una cartera de empresas elegibles.

**Se solicita:**

1. Construir sobre `perfil_reto` una consulta SQL que incluya `SELECT`, `WHERE`, condiciones múltiples enlazadas con `AND` y `ORDER BY`.
2. Establecer como criterios una razón corriente mayor o igual a 1 y un ROE positivo.
3. Ordenar el resultado por endeudamiento, de menor a mayor.
4. Determinar cuántas empresas satisfacen ambos criterios.

**Tiempo estimado:** 10 a 12 minutos.


In [0]:
CONSULTA_RETO = """
SELECT empresa,
       ROUND(razon_corriente, 2)     AS razon_corriente,
       ROUND(roe * 100, 1)           AS roe_pct,
       ROUND(endeudamiento * 100, 1) AS endeudamiento_pct
FROM perfil_reto
WHERE razon_corriente >= 1
  AND roe > 0
ORDER BY endeudamiento_pct ASC
"""

resultado_reto = duckdb.sql(CONSULTA_RETO).df()
display(resultado_reto)
print(f"Empresas que satisfacen ambos criterios: {len(resultado_reto)}")

,empresa,razon_corriente,roe_pct,endeudamiento_pct
0,AGROINDUSTRIAL LAREDO S.A.A.,1.10,9.0,34.8
1,CASA GRANDE SOCIEDAD ANONIMA ABIERTA,1.18,7.0,45.4
2,EMPRESA AGROINDUSTRIAL POMALCA S.A.A.,1.55,1.4,50.2


Empresas que satisfacen ambos criterios: 3


**Pregunta 7.** Según los criterios establecidos, ¿qué empresa presenta la condición financiera más favorable dentro del grupo analizado?

**Respuesta :** Con los datos disponibles hasta ahora, solo Casa Grande satisface simultáneamente razón corriente ≥ 1 y ROE positivo entre las empresas confirmadas — Cartavio queda excluida por su razón corriente de 0.96. Esto no significa que Casa Grande sea "la mejor" en términos absolutos (su ROE es menor), sino que es la única, de las dos confirmadas, que cumple ambos criterios de riesgo definidos por la consulta. El resultado definitivo depende de incorporar Laredo y Pomalca.

---


### Ejercicio 5 — Representación conjunta de la rentabilidad y del nivel de endeudamiento

Los gráficos de barras construidos en la Actividad 2 comparan una dimensión por vez. La decisión empresarial, sin embargo, rara vez depende de un solo indicador: una rentabilidad elevada sostenida sobre un endeudamiento igualmente elevado no equivale a una rentabilidad semejante obtenida con una estructura financiera conservadora, aun cuando ambas presenten el mismo ROE.

El gráfico de dispersión permite examinar dos dimensiones de manera simultánea y, mediante el tamaño del marcador, incorporar una tercera. Al situar el endeudamiento en el eje horizontal y el ROE en el eje vertical, cada cuadrante adquiere una lectura gerencial precisa: la combinación de rentabilidad elevada con endeudamiento reducido corresponde al perfil más favorable, mientras que la combinación inversa exige un examen detenido.

**Se solicita:**

1. Construir con Plotly un gráfico de dispersión que sitúe el endeudamiento en el eje horizontal y el ROE en el eje vertical.
2. Identificar cada punto con el nombre de la empresa correspondiente.
3. Representar el activo total mediante el tamaño del marcador.
4. Rotular los ejes y titular el gráfico de modo que resulte interpretable sin recurrir al código.

**Tiempo estimado:** 10 a 12 minutos.


In [0]:
fig_dispersión = go.Figure(go.Scatter(
    x=perfil_reto["endeudamiento"] * 100,
    y=perfil_reto["roe"] * 100,
    mode="markers+text",
    text=perfil_reto["empresa"],
    textposition="top center",
    marker=dict(
        size=perfil_reto["activo_total"] / perfil_reto["activo_total"].max() * 60 + 15,
        color=perfil_reto["roe"],
        colorscale="Tealrose",
        showscale=False,
        line=dict(width=1, color="white"),
    ),
    hovertemplate="%{text}<br>Endeudamiento: %{x:.1f}%<br>ROE: %{y:.1f}%<extra></extra>",
))
fig_dispersión.update_layout(
    title="Rentabilidad (ROE) vs. Endeudamiento — tamaño = Activo total",
    xaxis_title="Endeudamiento (%)",
    yaxis_title="ROE (%)",
    height=520,
    width=850,
    template="plotly_white",
)
fig_dispersión.show()


**Pregunta 8.** ¿Qué empresa muestra la combinación más favorable entre rentabilidad y nivel de endeudamiento?

**Respuesta:** Entre las dos empresas confirmadas, ninguna domina en ambas dimensiones a la vez: Casa Grande tiene menor endeudamiento (45.4%) pero también menor ROE (7.0%); Cartavio tiene mayor ROE (24.0%) pero también mayor endeudamiento (58.6%). El cuadrante más favorable (ROE alto con endeudamiento bajo) está vacío en este par — habrá que revisar si Laredo o Pomalca ocupan esa posición una vez completados sus datos.

---


### Retroalimentación final — Vinculación con el modelo DELTA Plus

Los cinco ejercicios anteriores se resolvieron mediante Python, DuckDB y Plotly sobre datos financieros oficiales. Corresponde ahora situar ese trabajo dentro del marco conceptual de la sesión. Responda de manera breve.

**Pregunta 9.** A partir del análisis realizado mediante Python, identifique qué dimensión de DELTA Plus se evidencia directamente en esta actividad: *Technology*, *Analytical Techniques* o ambas. Justifique su respuesta en un máximo de tres líneas.

**Respuesta:** Ambas. *Technology* se evidencia en el uso de Python, Polars, DuckDB y Plotly para extraer, transformar y visualizar datos oficiales; *Analytical Techniques* se evidencia en el cálculo de ratios financieros, la descomposición DuPont y las consultas condicionales — la tecnología fue el medio, las técnicas analíticas fueron el contenido.

---

**Pregunta 10.** *Technology.* ¿Qué tecnologías resultarían necesarias para que una de las empresas analizadas pudiera ejecutar de manera periódica y reproducible el análisis desarrollado en este laboratorio?

**Respuesta:** Una conexión automatizada (ETL) a las APIs de la SMV y el BCRP que se ejecute en cada cierre de ejercicio, una base de datos o data warehouse para almacenar las series históricas sin depender de una consulta manual cada vez, y un panel de BI (o un notebook programado, como este) que recalcule los ratios y regenere los gráficos automáticamente cuando lleguen datos nuevos.

---

**Pregunta 11.** *Analytical Techniques.* ¿En qué nivel de técnicas analíticas se sitúa el análisis realizado en esta sesión y qué técnica adicional podría incorporar la organización para mejorar sus decisiones?

**Respuesta:** Se sitúa en un nivel descriptivo-diagnóstico: se miden resultados (ratios) y se explican mediante descomposición (DuPont), pero no se proyecta ni se prescribe nada. Una técnica adicional razonable sería análisis de series de tiempo sobre varios ejercicios fiscales para proyectar tendencias de rentabilidad y liquidez, en vez de comparar un solo año de forma estática.

---


## Conclusiones

Registre tres conclusiones técnicas y de negocio:

1. **Verificación técnica de la rentabilidad (Identidad DuPont):** Con los datos oficiales confirmados de las cuatro principales empresas del sector azucarero (ejercicio 2024), la identidad DuPont se verifica matemáticamente en todas (diferencia < 0.001 entre ROA real y ROA calculado). Esto confirma la consistencia de los datos y demuestra que la rentabilidad sobre activos en este sector se explica de forma rigurosa por la combinación del margen neto y la rotación de activos, sin variables ocultas.
2. **Trade-off competitivo entre riesgo y retorno:** Ninguna empresa domina simultáneamente en rentabilidad, liquidez y endeudamiento. **Cartavio** asume un perfil agresivo (lidera en ROE con 24.0% y rotación con 0.58, pero es la única con déficit de liquidez de 0.96 y mayor endeudamiento con 58.6%). **Laredo** y **Casa Grande** adoptan posturas conservadoras, priorizando la solidez patrimonial a costa de un menor retorno. **Pomalca** evidencia que la liquidez ociosa (1.55) combinada con ineficiencia operativa destruye el perfil competitivo (ROE marginal del 1.4%).
3. **Límites del análisis financiero en la madurez analítica:** La metodología aplicada permitió construir un perfil financiero reproducible utilizando motores modernos (Polars y DuckDB), lo que evidencia rigor en el tratamiento de los datos. Sin embargo, los estados financieros son insuficientes para evaluar dimensiones organizacionales del marco DELTA Plus (*Analysts, Leadership, Targets, Enterprise*). No es posible determinar conclusivamente el estadio de madurez analítica de una empresa sin evidencia cualitativa sobre sus procesos internos de toma de decisiones.

## Referencias oficiales

- Superintendencia del Mercado de Valores. Portal de Datos Abiertos y servicio de información financiera.
- Banco Central de Reserva del Perú. Series estadísticas mensuales.